### PR TITLE
[ACR] Fix failing samples

### PR DESCRIPTION
**Pull request by YalinLi0312**
* Added pre-requisites for samples, and corrected comments.
* Updated `test.yml` to enable samples running in live pieplines, which resolves: https://github.com/Azure/azure-sdk-for-python/issues/27035

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py [hunk 1]
```diff
@@ -18,8 +20,13 @@
...
 from ._generated.models import OCIManifest
 
+from azure.mgmt.containerregistry import ContainerRegistryManagementClient
+from azure.mgmt.containerregistry.models import ImportImageParameters, ImportSource, ImportMode
+from azure.identity import DefaultAzureCredential, AzureAuthorityHosts, ClientSecretCredential
+from azure.identity.aio import DefaultAzureCredential as AsyncDefaultAzureCredential
+
 if TYPE_CHECKING:
-    from typing import List, Dict, IO
+    from typing import Dict, IO
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py:29]: These imports should be moved to line 13

File: sdk/containerregistry/azure-containerregistry/samples/async_samples/sample_base_async.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_base_async.py
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/samples/async_samples/sample_base_async.py:10]: Why does this file exist? We shouldn't need a base for samples?

File: sdk/containerregistry/azure-containerregistry/samples/sample_base.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+FILE: sample_base.py
+
+DESCRIPTION:
+    This is the base class for sync samples. It imports some images that could be used in samples.
+"""
+import os
+from dotenv import find_dotenv, load_dotenv
+from azure.containerregistry._helpers import _import_image, _get_authority,_get_audience, _get_credential
+
+class SampleBase(object):
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/samples/sample_base.py:19]: We shouldn't use a base class for samples - it makes it very awkward to read and copy+paste code from the samples

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py [hunk 4]
```diff
@@ -150,3 +158,73 @@ def _validate_digest(data, digest):
     # type: (IO, str) -> bool
     data_digest = _compute_digest(data)
     return data_digest == digest
+
+def _import_image(authority: str, repository: str, tags: Optional[List[str]]) -> None:
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py:162]: These methods are test utilities and shouldn't live in the SDK.

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py [hunk 5]
```diff
@@ -18,8 +20,13 @@
 from azure.core.exceptions import ServiceRequestError
 from ._generated.models import OCIManifest
 
+from azure.mgmt.containerregistry import ContainerRegistryManagementClient
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py:23]: This SDK doesn't have a dependency on the mgmt SDK does it?

Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py:23]: It does, the mgmt client is used to import images.

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_helpers.py:29]: I have a separate PR for hint style and will get rid of TYPE_CHECKING, similar to one you have reviewed for ApppConfig. 
Update: the PR is in draft, will be ready for review after the mypy PR get merged.

File: sdk/containerregistry/azure-containerregistry/samples/load_registry.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    import_params = ImportImageParameters(mode=ImportMode.Force, source=import_source, target_tags=tags)
+
+    result = mgmt_client.registries.begin_import_image(
+        rg_name,
+        registry_name,
+        parameters=import_params,
+    )
+
+    while not result.done():
+        pass
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/samples/load_registry.py:66]: You shouldn't have to write a loop like this.
You should be able to use `result.wait()`

File: sdk/containerregistry/azure-containerregistry/samples/load_registry.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/samples/load_registry.py:2]: Looks like this file name should be prefixed with `sample_`?
Also - there doesn't seem to be an async version.

File: sdk/containerregistry/azure-containerregistry/samples/async_samples/sample_delete_images_async.py [hunk 8]
```diff
@@ -17,36 +17,51 @@
...
+"""
 import asyncio
-from dotenv import find_dotenv, load_dotenv
 import os
-
+from dotenv import find_dotenv, load_dotenv
 from azure.containerregistry import ArtifactManifestOrder
 from azure.containerregistry.aio import ContainerRegistryClient
-from azure.identity.aio import DefaultAzureCredential
+from azure.containerregistry._helpers import _get_authority, _get_audience, _get_credential
```
Comment (annatisch) [inline sdk/containerregistry/azure-containerregistry/samples/async_samples/sample_delete_images_async.py:28]: The samples shouldn't demonstrate use of private methods. These should probably just be in each sample.

**Review by annatisch — APPROVED** [review]

**Pull request merged**